core/templates/domain/exploration/editable-exploration-backend-api.service.spec.ts

```diff
@@ -95,180 +73,182 @@ describe('Editable exploration backend API service', function() {
       user_email: 'test@example.com',
       version: 1
     };
-  }));
+  });
 
   afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
+    httpTestingController.verify();
   });
 
   it('should successfully fetch an existing exploration from the backend',
-    function() {
+    fakeAsync(() => {
       var successHandler = jasmine.createSpy('success');
       var failHandler = jasmine.createSpy('fail');
 
-      $httpBackend.expect('GET', '/createhandler/data/0').respond(
-        sampleDataResults);
-      EditableExplorationBackendApiService.fetchExploration('0').then(
+      editableExplorationBackendApiService.fetchExploration('0').then(
         successHandler, failHandler);
-      $httpBackend.flush();
+      let req = httpTestingController.expectOne('/createhandler/data/0');
+      expect(req.request.method).toEqual('GET');
+      req.flush(sampleDataResults);
+      flushMicrotasks();
 
       expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
       expect(failHandler).not.toHaveBeenCalled();
-    }
+    })
   );
 
   it('should fetch and apply the draft of an exploration',
-    function() {
+    fakeAsync(() => {
       var successHandler = jasmine.createSpy('success');
```

Comment:
Please use `let` here and in other places below.

Reply:
Done